pkg/operator2/configsync.go

```diff
@@ -134,25 +134,33 @@ func newIDPSyncData() idpSyncData {
 // AddSecret initializes a sourceData object with proper data for a Secret
 // and adds it among the other secrets stored here
 // Returns the path for the Secret
-func (sd *idpSyncData) AddSecret(index int, secretName configv1.SecretNameReference, key string) string {
+func (sd *idpSyncData) AddSecret(index int, secretName configv1.SecretNameReference, key string, optional bool) string {
+	if optional && len(secretName.Name) == 0 {
```

Comment:
Validation should precede these additions, here I'd personally rather only do
```
if len(secretName.Name) == 0 {
    return ""
}
```
It's for a simple reason - you'll need to return errors, too, which is something you're not doing here. Otherwise the problem we have now when optional field is unset would now still be present for required fields with unset values.

Proposal:
Use just the condition mentioned above. That will effectively make all such fields optional. I prepared some preliminary validation code for our CRDs for origin, I need to re-check it and then post it, then I'll get to validation on the operator side.